### PR TITLE
Style item::tname (part 1 of 2)

### PR DESCRIPTION
**Pull request by sethsimon**
I plan to work on #19240, but the current style is too distracting.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -2155,109 +2155,113 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
...
             burntext = pgettext( "burnt adjective", "burnt " );
         }
     }
 
-    const std::map<std::string, std::string>::const_iterator iname = item_vars.find("name");
-    std::string maintext = "";
-    if (corpse != NULL && typeId() == "corpse" ) {
+    const std::map<std::string, std::string>::const_iterator iname = item_vars.find( "name" );
+    std::string maintext;
+    if( corpse != nullptr && typeId() == "corpse" ) {
```
Comment (codemime) [inline src/item.cpp]: This condition can be replaced with `is_corpse()`.

**Comment by codemime** [comment]
`item::type_name()` duplicates a chunk of code from `item::tname()`. It would be good if you could eliminate the duplication in this PR (whilst the duplication is still verbatim).

**Comment by sethsimon** [comment]
Thanks, I didn't realize that. Since I've already changed those lines, I changed them again to remove the duplication.

File: src/item.cpp [hunk 2]
```diff
@@ -2155,109 +2155,97 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
...
+                maintext = string_format( pgettext( "item name", "%s of %s" ), label( quantity ).c_str(),
+                                          contents.front().tname( contents.front().charges, with_prefix ).c_str() );
+            } else {
+                maintext = string_format( pgettext( "item name", "%s of %s" ), label( quantity ).c_str(),
+                                          contents.front().tname( quantity, with_prefix ).c_str() );
+            }
         } else {
-            maintext = string_format( pgettext( "item name", "%s with %s" ), label(quantity).c_str(), contents.front().tname( quantity, with_prefix ).c_str());
+            maintext = string_format( pgettext( "item name", "%s with %s" ), label( quantity ).c_str(),
+                                      contents.front().tname( quantity, with_prefix ).c_str() );
```
Comment (codemime) [inline src/item.cpp]: Nitpicking: these assignments look very alike. Could be nicely simplified.

**Comment by sethsimon** [comment]
>Nitpicking: these assignments look very alike. Could be nicely simplified.

Fixed.

**Pull request merged**